### ISSUE TITLE
PoS add re-delegation

### ISSUE THREAD

**Issue by tzemanovic** [anoma/anoma#637]
Add a transaction for re-delegation (using a version with longer delay, which has to wait for unbonding epoch, but simpler to implement than faster re-delegation).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201762884076015) by [Unito](https://www.unito.io)

**Comment by tzemanovic** [comment]
There's a potential problem with slashing that's applied after re-delegation, because the re-delegated amount would be unbonded and bonded to the new target at unbonding offset, but unlike tokens that are unbonded for withdrawal, which are being slashed when they are being withdrawn, there's no further action required with the current re-delegation design.

When we re-visit re-delegation, we should add spec of it as it has to be accepted by the validity predicate and we should re-visit the integration spec in https://dev.anoma.net/master/explore/design/ledger/pos-integration.html?highlight=redelegate#delegator-transactions - this is not very clear, because we're re-using `delegate` function definition, which also transfers tokens, but re-delegation keeps all the tokens in PoS.

**Comment by cwgoes** [comment]
At the point the re-delegated amount is bonded to the new target, can we check if there were any slashes since discovered that need to be applied?

(this may require some sort of queue, but the processing required should be low and thus safe, I think?)

**Comment by tzemanovic** [comment]
Yeah, we can do that with some queue that is processed by the protocol - it will need to update data in the re-delegation target if and when some slash(es) for the source are found
